jaggeryapps/admin/serviceproviders/custom/controllers/custom/edit_finish.jag

```diff
@@ -522,122 +522,135 @@ function addApp(){
-        fileContentObj.setContent(bannerInputStream);
-        fileContentObj.setFileName(bannerFile.getName());
-        bannerId = String(apiProviderImplObj.uploadImage(fileContentObj));
-
-        bannerInputStream.close();
-    }else if(storeAppType === WELL_KNOWN_APP){
+    // If there is no existing banner for the application,
+    //check whether default type application
+    if (!bannerId && storeAppType === WELL_KNOWN_APP) {
 
-      // This can be a well known application.
-      // The image should be taken from the file system.
+        // This can be a well known application.
+        // The image should be taken from the file system.
 
-      var applicationType = request.getParameter("spType");
-      var bannerFileName = getImageFilePathForWellKnownAppType(applicationType);
+        var applicationType = request.getParameter("spType");
+        var bannerFileName = getImageFilePathForWellKnownAppType(applicationType);
 
-      var bannerFile;
-      var bannerInputStream;
-      try {
-        bannerFile = new File(bannerFileName);
-        bannerFile.open("r");
+        var bannerFile;
+        var bannerInputStream;
+        try {
+            bannerFile = new File(bannerFileName);
+            bannerFile.open("r");
+
+            bannerInputStream = bannerFile.getStream().getStream();
+            fileContent.setContent(bannerInputStream);
+            fileContent.setFileName(applicationType + ".png");
+            bannerId = String(apiProviderImplObj.uploadImage(fileContent));
+        } catch (err) {
+            log.error("Can't read and upload banner file for the application.", err);
+        } finally {
+            if (bannerInputStream) {
+                bannerInputStream.close();
+            }
 
-        bannerInputStream = bannerFile.getStream().getStream();
-        fileContent.setContent(bannerInputStream);
-        fileContent.setFileName(applicationType + ".png");
-        bannerId = String(apiProviderImplObj.uploadImage(fileContent));
-      }catch(err) {
-        log.error("Can't read and upload banner file for the application.", err);
-      }finally {
-        if(bannerInputStream){
-          bannerInputStream.close();
+            if (bannerFile) {
+                bannerFile.close();
+            }
         }
-
-        if(bannerFile){
-          bannerFile.close();
+    } else {
+        //  check whether an image binary comes with the request.
+        var bannerFile = request.getFile("bannerFile");
+        if (bannerFile != null) {
+            try {
+                var fileContentObj = new Packages.org.wso2.carbon.appmgt.api.model.FileContent;
+                var bannerInputStream = bannerFile.getStream().getStream();
+                fileContentObj.setContent(bannerInputStream);
+                fileContentObj.setFileName(bannerFile.getName());
+                bannerId = String(apiProviderImplObj.uploadImage(fileContentObj));
+
+                bannerInputStream.close();
```

Comment:
This is closed in the finally block properly right. This line should be removed.